src/doc/markdown/page111_developing_with_react_context.md

```diff
@@ -0,0 +1,162 @@
+# Developing with React Context
+## New Web Directories
+## `components`
+#### What lives inside this directory?
+* Reusable React components
+* *Examples:*
+    * Toggle
+    * Button
+    * InputElement
+
+## `containers`
+#### What lives inside this directory?
+* An element likely to be used only once in the interface that consists of multiple components
+* *Examples:*
+    * BotDetails
+    * SettingsPanel
+    * MissionButtonRow
+
+## `context`
+#### What lives inside this directory?
+* Contexts for the app. These hold state for different parts of the system in a global fashion.
+* *Examples:*
+    * GlobalContext
+    * PodContext
+    * HubContext
+
+## `utils`
+#### What lives inside this directory?
+* Functions that abstract data processing away from React code
+* Eventually we will migrate `src/web/shared/Utilities.tsx` functions into this directory
+
+# Context Structure
+### `GlobalContext`
+* Contains general data pertaining to the view state of the interface
+* *Examples:*
+    * `selectedPodElement`
+    * `showHubDetails`
+    * `hubAccordionStates`
+#### `PodContext`
+* Encapsulates `HubContext` and `BotContext`
+* Provides a location to store pod-wide data
+##### `HubContext`
+* Stores the `PortalHubStatus`
+##### `BotContext`
+* Coming soon
+
+## General Context Strcuture
+1. Imports
+2. Interfaces
+3. Constant Variables and Enums
+4. Context Instantiation
+5. Reducer Function
+6. Reducer Helper Functions
+7. ContextProvider Component
+
+
+#### 1. Imports
+#### 2. Interfaces
+##### Required Interfaces
+* ContextType
+    * Identifies the properties and types that will be stored in the context
+    * Think of this as the state interface
+```
+// Example
+interface GlobalContextType {
+    clientID: string,
+    controllingClientID: string
+    selectedPodElement: SelectedPodElement
+    ...
+}
+```
+* Action
+    * Lays out the properties and types of the action object passed to the reducer function. `type` is the only required property.
+    * The optional properties are used to pass data to specific reducer helper functions
+```
+// Example
+interface Action {
+    type: string,
+    clientID?: string,
+    hubAccordionName?: string
+}
+```
+* ContextProviderProps
+    * Satisfies TypeScript's requirements for accepting a React Component as a prop
+```
+// Example
+interface GlobalContextProviderProps {
+    children: ReactNode
+}
+```
+
+#### 3. Constant Variables and Enums
```

Comment:
I didn't add a description because I didn't think it would add much more than what the title already says, but I can add a short blurb if you would like for consistency with the other sections